pkg/storage/unified/sql/queries.go

```diff
@@ -306,6 +307,7 @@ func (r sqlGetHistoryRequest) Validate() error {
 type sqlPruneHistoryRequest struct {
 	sqltemplate.SQLTemplate
 	Key          *resource.ResourceKey
+	Generation   bool // partion by generation
```

Comment:
nit; naming is a bit confusing given it's a bool not an int. Maybe `PartitionByGeneration` instead of `Generation` ?